File: 367 isPerfectSquare/367.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if(num==1)return 1;
        int left=1;
        int right=num-1;
        while(left<=right){
            int middle=(right+left)/2;
            if((long long)middle*middle==num){
                return true;
            }
            if((long long)middle*middle>num){
                right=middle-1;
            }
            else{
                left=middle+1;
            }
        }
        return false;
    }
};

File: 367 isPerfectSquare/367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num==1:
            return True
        left=1
        right=num-1
        while left<=right:
            middle=(left+right)//2
            if middle*middle==num:
                return True
            elif middle*middle > num:
                right=middle-1
            else:
                left=middle+1
        return False